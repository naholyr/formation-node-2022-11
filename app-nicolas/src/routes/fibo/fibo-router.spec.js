import request from "supertest";

const { app } = await import("../../app.js");

describe("App", () => {
  describe("GET /fibo/:number", () => {
    it("should call fibo()", async () => {
      const response = await request(app).get("/fibo/sync/10");
      expect(response.status).toEqual(200);
      expect(response.body).toMatchObject({ output: 55 });
      expect(response.body).toMatchSnapshot();
    });
  });
});
