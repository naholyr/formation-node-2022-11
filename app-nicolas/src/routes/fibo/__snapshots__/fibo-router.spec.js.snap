// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App GET /fibo/:number should call fibo() 1`] = `
{
  "input": 10,
  "output": 55,
}
`;
